owncloudApp/src/main/java/com/owncloud/android/workers/UploadFileFromContentUriWorker.kt

```diff
@@ -157,14 +199,80 @@ class UploadFileFromContentUriWorker(
 
     private fun uploadDocument() {
         val client = getClientForThisUpload()
-        val requestBody = ContentUriRequestBody(appContext.contentResolver, contentUri).also {
-            it.addDatatransferProgressListener(this)
+
+        val cacheFile = File(cachePath)
+        mimeType = cacheFile.extension
+        fileSize = cacheFile.length()
+
+        val getStoredCapabilitiesUseCase: GetStoredCapabilitiesUseCase by inject()
+        val capabilitiesForAccount = getStoredCapabilitiesUseCase.execute(
+            GetStoredCapabilitiesUseCase.Params(
+                accountName = account.name
+            )
+        )
+        val isChunkingAllowed = capabilitiesForAccount != null && capabilitiesForAccount.isChunkingAllowed()
+        Timber.d("Chunking is allowed: %s", isChunkingAllowed)
+
+        if (isChunkingAllowed && fileSize > CHUNK_SIZE) {
+            uploadChunkedFile(client)
+        } else {
+            uploadPlainFile(client)
+        }
+        removeCacheFile()
+    }
+
+    private fun uploadPlainFile(client: OwnCloudClient) {
+        uploadFileOperation = UploadFileFromFileSystemOperation(
+            localPath = cachePath,
+            remotePath = uploadPath,
+            mimeType = mimeType,
+            lastModifiedTimestamp = lastModified,
+            requiredEtag = null,
+        ).also {
+            it.addDataTransferProgressListener(this)
+        }
+
+        val result = executeRemoteOperation { uploadFileOperation.execute(client) }
+
+        if (result == Unit && behavior == UploadBehavior.MOVE) {
+            removeLocalFile()
         }
+    }
```

Comment:
This is the code that should be placed just after copying the file to the cache 🍻